DiscordBot/Program.cs

```diff
@@ -40,9 +47,83 @@ private static async Task MainAsync()
             commands.RegisterCommands<EventCommands>();
 
             discord.UseInteractivity(new InteractivityConfiguration());
+            discord.MessageReactionAdded += async (client, eventArguments) =>
+            {
+                await OnMessageReactionAdded(client, eventArguments, eventsSheetsService);
+            };
 
             await discord.ConnectAsync();
             await Task.Delay(-1);
         }
+
+        private static async Task OnMessageReactionAdded(
+            DiscordClient client,
+            MessageReactionAddEventArgs eventArguments,
+            IEventsSheetsService eventsSheetsService)
+        {
+            if (eventArguments.User == client.CurrentUser)
+            {
+                return;
+            }
+
+            var discordEvent = await GetEventFromMessageIdOrDefaultAsync(eventArguments.Message.Id, eventsSheetsService);
+            if (discordEvent == null)
+            {
+                return;
+            }
+
+            await ProcessReaction(client, eventArguments, discordEvent, eventsSheetsService);
+
+            await UpdateSignupSheet(eventArguments, discordEvent, eventsSheetsService);
```

Comment:
Can the two things above be done in parallel?

Reply:
No because the updating the signup sheet obtains the new signup information from the sheet, meaning the sheet needs to be updated first.

Reply:
Ah okay 🙂